File: why_LeetCode/Day09/C++/Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        add(digits,1);
        return digits;
    }

private:
void add(vector<int>& digits, int digit)
{
    int c=digit; //½øÎ»
    for(auto i=digits.rbegin();i!=digits.rend();i++)
    {
        *i+=c;
        c=*i/10;
        *i%=10;
    }

    if(c>0) digits.insert(digits.begin(),c);
}

};

File: why_LeetCode/Day09/Python/Plus One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        c=1
        
        for i in range(len(digits)-1,-1,-1):
            digits[i]+=c
            c=digits[i]//10
            digits[i]%=10
        
        if c>0:
            digits.insert(0,c)
        
        return digits